### PR TITLE
Format resource diffs

### LINKED ISSUE #973: kops dry-run shows "<resource> -> <resource>"
I had a modified resource on production and when executing the dry-run shows as...

```
  IAMRole               iamRole/nodes.production-a-cluster.k8s-production.photobox.com
        RolePolicyDocument       <resource> -> <resource>
```

This makes it very difficult to identify what changes are going to be applied.

@yissachar mentioned patching https://github.com/kubernetes/kops/blob/472c443d1b383b93c7f24faf205bc94da8eefe94/upup/pkg/fi/dryrun_target.go#L288%23291

### PR DESCRIPTION
**Pull request by justinsb**
Fix #973

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/997)

### CODE REVIEW

**Comment by justinsb** [comment]
I don't think it's perfect - the logic is surprisingly tricky - but I think it's a good start.

File: upup/pkg/fi/dryrun_target.go [hunk 1]
```diff
@@ -186,14 +190,22 @@ func (t *DryRunTarget) PrintReport(taskMap map[string]Task, out io.Writer) error
...
+
+							if !ignored && description == "" {
 								description = fmt.Sprintf(" %v -> %v", ValueAsString(fieldValA), ValueAsString(fieldValE))
 							}
 						}
 						if ignored {
 							continue
 						}
-						changeList = append(changeList, fmt.Sprintf("%-20s\t%s", valC.Type().Field(i).Name, description))
+						changeList = append(changeList, change{FieldName: valC.Type().Field(i).Name, Description: description})
```
Comment (chrislovecnm) [inline upup/pkg/fi/dryrun_target.go]: should we be checking `valC.Type().Field(i).Name` to see if the value exists?  Seems like a lot of possibilities for a panic.

File: upup/pkg/fi/utils/diff.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+package utils
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+)
+
+func FormatDiff(lString, rString string) string {
```
Comment (chrislovecnm) [inline upup/pkg/fi/utils/diff.go]: really important stuff, is a unit test a good idea?

**Comment by chrislovecnm** [comment]
@justinsb unit tests?

File: upup/pkg/fi/dryrun_target.go (hunk 1 above)
Comment (justinsb) [inline upup/pkg/fi/dryrun_target.go]: Go reflection is not great, but we've already checked valC.Field(i), so I think this is right...

File: upup/pkg/fi/utils/diff.go (hunk 2 above)
Comment (justinsb) [inline upup/pkg/fi/utils/diff.go]: Probably.  I'll do a quickie because I'm not sure I love the formatting yet (e.g. does the - or + come first?)

**Comment by justinsb** [comment]
Added unit tests & fixed up some issues; merging

**Pull request merged**